onnxruntime/core/graph/graph_flatbuffers_utils.cc

```diff
@@ -0,0 +1,406 @@
+  std::vector<TYPE> NAME(SRC_DATA.size()); \
+  std::copy(SRC_DATA.cbegin(), SRC_DATA.cend(), NAME.begin());
+
+Status SaveAttributeOrtFormat(flatbuffers::FlatBufferBuilder& builder,
+                              const AttributeProto& attr_proto,
+                              flatbuffers::Offset<fbs::Attribute>& fbs_attr,
+                              const onnxruntime::Graph* graph) {
+  auto name = builder.CreateString(attr_proto.name());
+  auto doc_string = builder.CreateString(attr_proto.doc_string());
+  auto type = static_cast<fbs::AttributeType>(attr_proto.type());
+  switch (type) {
+    case fbs::AttributeType_FLOAT: {
+      GET_FBS_ATTR(builder, type, f, attr_proto.f());
+    } break;
+    case fbs::AttributeType_INT: {
+      GET_FBS_ATTR(builder, type, i, attr_proto.i());
+    } break;
+    case fbs::AttributeType_STRING: {
+      auto s = builder.CreateString(attr_proto.s());
+      GET_FBS_ATTR(builder, type, s, s);
+    } break;
+    case fbs::AttributeType_TENSOR: {
+      flatbuffers::Offset<fbs::Tensor> fbs_tensor;
+      ORT_RETURN_IF_ERROR(
+          experimental::utils::SaveInitializerOrtFormat(builder, attr_proto.t(), fbs_tensor));
+      GET_FBS_ATTR(builder, type, t, fbs_tensor);
+    } break;
+    case fbs::AttributeType_GRAPH: {
+      ORT_RETURN_IF_NOT(nullptr != graph, "GetAttributeOrtFormat, graph is null");
+      flatbuffers::Offset<fbs::Graph> fbs_graph;
+      ORT_RETURN_IF_ERROR(graph->SaveToOrtFormat(builder, fbs_graph));
+      GET_FBS_ATTR(builder, type, g, fbs_graph);
+    } break;
+    case fbs::AttributeType_FLOATS: {
+      GET_DATA_VEC(float, floats_vec_, attr_proto.floats());
+      auto floats = builder.CreateVector(floats_vec_);
+      GET_FBS_ATTR(builder, type, floats, floats);
+    } break;
+    case fbs::AttributeType_INTS: {
+      GET_DATA_VEC(int64_t, ints_vec_, attr_proto.ints());
+      auto ints = builder.CreateVector(ints_vec_);
+      GET_FBS_ATTR(builder, type, ints, ints);
+    } break;
+    case fbs::AttributeType_STRINGS: {
+      GET_DATA_VEC(std::string, strings_vec_, attr_proto.strings());
+      auto strings = builder.CreateVectorOfStrings(strings_vec_);
+      GET_FBS_ATTR(builder, type, strings, strings);
+    } break;
+    case fbs::AttributeType_TENSORS: {
+      std::vector<flatbuffers::Offset<fbs::Tensor>> fbs_tensors_vec;
+      fbs_tensors_vec.reserve(attr_proto.tensors().size());
+      for (const auto& tensor : attr_proto.tensors()) {
+        flatbuffers::Offset<fbs::Tensor> fbs_tensor;
+        ORT_RETURN_IF_ERROR(
+            experimental::utils::SaveInitializerOrtFormat(builder, tensor, fbs_tensor));
+        fbs_tensors_vec.push_back(fbs_tensor);
+      }
+      auto tensors = builder.CreateVector(fbs_tensors_vec);
+      GET_FBS_ATTR(builder, type, tensors, tensors);
+    } break;
+    default:
+      break;
+  }
+
+  return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "GetAttributeOrtFormat - Unsupported type: ", type);
+}
```

Comment:
Not a fan of unnecessarily hiding a 'return' in a macro. if this error return is in the 'default:' case we can just return Status::OK() outside of the switch can't we, and remove the 'return' from the macro?